pkgs/tools/system/osquery/default.nix

```diff
@@ -0,0 +1,79 @@
+{ pkgs }:
+
+with pkgs;
+let
+  buildStdenv = overrideCC stdenv llvmPackages.clangUseLLVM;
+  opensslArchive =
+    let
+      # https://github.com/osquery/osquery/blob/master/libraries/cmake/formula/openssl/CMakeLists.txt#L3-L4
+      version = "1.1.1q";
+      sha256 = "d7939ce614029cdff0b6c20f0e2e5703158a489a72b2507b8bd51bf8c8fd10ca";
+    in
+    fetchurl {
+      inherit sha256;
+      url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
+    };
+in
+buildStdenv.mkDerivation rec {
+  pname = "osquery";
+  version = "5.5.1";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+
+    fetchSubmodules = true;
+    sha256 = "sha256-Q6PQVnBjAjAlR725fyny+RhQFUNwxWGjLDuS5p9JKlU=";
+  };
+
+  patches = [
+    ./Remove-git-reset.patch
+    ./Use-locale.h-instead-of-removed-xlocale.h-header.patch
+    ./Remove-circular-definition-of-AUDIT_FILTER_EXCLUDE.patch
+    ./Remove-system-controls-table.patch
```

Comment:
I think this is fine for now. I'd like to see two things, if you make sense:

1. An upstream issue. I know osquery loves to pin their glibc and other dependencies, but they'll have to cut over at some point, right?
2. Some documentation, maybe in the description, about differences from upstream (namely, this one).

Reply:
Absolutely agree, will open that issue and reference it here.

Reply:
Opened https://github.com/osquery/osquery/issues/7823, added a reference in the package, and updated the package description in ccdb07a2490